components/support/nimbus-fml/src/parser.rs

```diff
@@ -179,6 +186,35 @@ impl Parser {
         Ok(merged)
     }
 
+    fn inline_manifest_resources(
+        &self,
+        path: &FilePath,
+        manifest: &mut ManifestFrontEnd,
+    ) -> Result<()> {
+        for feature in manifest.features.values_mut() {
+            let as_typed = &feature.examples;
+            let mut inlined = Vec::with_capacity(as_typed.len());
+            for example in as_typed {
+                inlined.push(example.inline(&self.files, path)?);
+            }
+            feature.examples = inlined;
```

Comment:
You can map with a result-returning function and then collect into a `Result<Vec<_>, _>` here, which will return either `Ok(vec)` or `Err(first_error)`:

```suggestion
            feature.examples = as_typed
                .iter()
                .map(|ex| ex.inline(&self.files, path))
                .collect::<Result<Vec<_>, _>()?;
```

Reply:
Neat!